scripts/sync-gsuite.js

```diff
@@ -0,0 +1,233 @@
+function putInGsuiteGroups(obj) {
+  return "wip"
+
+  /* For each circle which has a gsuite_id field set,
+   * take a list of members.
+   * For each member, execute API call.
+   *
+   *
+   * Blocked by policy (about people 'consenting' to receive mandatory email)
+   */
+
+  if(obj.allocated){
+
+    const result = core.query(obj.recoveryEmail).bodies
+    const users = await User.findAll({// FIXME or bodies?
+      where: { gsuite_id: obj.primaryEmail }
+    });
+
+    if(users){
+      for (const grp of result.gsuite_id) { //FIXME there must already be a mapping in core between body and gsuite
+        const data = {
+          "primaryEmail": obj.primaryEmail,
+          "group": grp
+        }
+        runGsuiteOperation(gsuiteOperations.addUserInGroup, data);
+        sleep(2) ; //rate limiting
+      }
+    }
+  }
+}
+
+//gsuite_obj.forEach( putInGsuiteGroups )
+
+// HERE is the part where we do the MANUAL PATCH
+console.log("")
+console.log("")
+gsuite_obj.filter(obj => obj.allocated===false).forEach( obj => console.log(`${obj.primaryEmail}'s secondary email on my aegee is not ${obj.recoveryEmail}`)) ;
+console.log("")
+console.log("Total number of unallocated objects:")
+console.log(gsuite_obj.filter(obj => obj.allocated===false).length)
+
+//HELPER For our manual patch: heuristic by surname
+async function findMyAEGEEuserBySurname(obj) {
+
+  const users = await User.findAll({
+    where: { last_name: obj.g_surname } //FIXME use "like"
+  });
+
+  console.log(users)
+
+//try match via surname
+  if(users.length > 0 ){
+
+    obj.m_possible_users = users
+    obj.allocated=true
+
+  }else{ //for our hungarian people or the others...
+    const users = await User.findAll({
+      where: { first_name: obj.g_surname } //FIXME use "like"
+    });
+    console.log(users)
+
+    //try match via surname
+    if(users.length > 0 ){
+
+      obj.m_possible_users = users
+      obj.allocated=true
+
+    }
+  }
+}
+
+gsuite_obj.filter(obj => obj.allocated===false).forEach( findMyAEGEEuserBySurname )
+console.log("")
+console.log("")
+console.log("heuristic: surname checking")
+
+console.log("easy guess (only one possibility):")
+gsuite_obj.filter(obj => obj.m_possible_users.length===1).forEach( obj => { console.log(`${obj.primaryEmail} could be the following: ${obj.m_possible_users}`) ; console.log(obj.m_possible_users[0]) }) ;
+console.log("easy guess total: ", gsuite_obj.filter(obj => obj.m_possible_users.length===1).length)
+
+console.log("harder guess")
+gsuite_obj.filter(obj => obj.m_possible_users.length>1).forEach( obj => { console.log(`${obj.primaryEmail} could be the following: ${obj.m_possible_users}`) ; console.log(obj.m_possible_users[0]) }) ;
+console.log("harder guess total: ", gsuite_obj.filter(obj => obj.m_possible_users.length>1).length)
+
+console.log("")
+console.log("")
+
+
+//LAST PART: create gsuite account for all members in MyAEGEE
```

Comment:
We don't have to do this part anymore, since we will include it in the profile update page

Reply:
great, I can delete it